test/e2e/clusters/clusters.go

```diff
@@ -0,0 +1,256 @@
+		clusterName = kindClusterName
+	case DistroK3d:
+		clusterName = k3dClusterName
+	}
+	tempKubeconfigFile, err := ioutil.TempFile("", clusterName+"*.yaml")
+	if err != nil {
+		return "", err
+	}
+
+	// Set the KUBECONFIG env var to the new file
+	err = os.Setenv("KUBECONFIG", tempKubeconfigFile.Name())
+	if err != nil {
+		return "", err
+	}
+
+	// Create the cluster
+	switch distroToUse {
+	case DistroKind:
+		err = createKindCluster(tempKubeconfigFile.Name())
+		if err != nil {
+			return "", err
+		}
+	case DistroK3d:
+		err = createK3dClusterUsingCurrentKubeconfig()
+		if err != nil {
+			return "", err
+		}
+	}
+	return tempKubeconfigFile.Name(), nil
+}
+
+// DeleteClusterAndTemporaryKubeconfig is intended to be deferred at the end of TestMain execution so we can clean up
+// the cluster and temp file that we made.
+func DeleteClusterAndTemporaryKubeconfig(distroToUse DistroToUse, tempKubeconfigFilePath string) error {
+	// Defer the temp file delete so it always happens no matter what
+	defer func(name string) {
+		_ = os.Remove(name)
+	}(tempKubeconfigFilePath)
+
+	// Delete the cluster
+	switch distroToUse {
+	case DistroKind:
+		err := deleteKindCluster(tempKubeconfigFilePath)
+		if err != nil {
+			return err
+		}
+	case DistroK3d:
+		err := deleteK3dCluster()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// TryValidateClusterIsRunning establishes a valid connection to the currently configured cluster, then returns,
+// throwing an error if anything went wrong.
+func TryValidateClusterIsRunning() error {
+	clientSet, err := GetClientSet()
+	if err != nil {
+		return fmt.Errorf("unable to connect to the cluster: %w", err)
+	}
+	pods, err := clientSet.CoreV1().Pods("kube-system").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to list running pods: %w", err)
+	}
+	if len(pods.Items) < 1 {
+		return fmt.Errorf("established connection to cluster, but no pods were found")
+	}
+
+	return nil
+}
+
+// GetConfig gets the currently configured Kubeconfig, or throws an error if something went wrong.
+func GetConfig() (*rest.Config, error) {
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+// GetClientSet establishes a connection with the currently configured k8s cluster and returns the ClientSet object
+// that is needed to interact with it.
+func GetClientSet() (*kubernetes.Clientset, error) {
+	config, err := GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientSet, nil
+}
+
+// createKindCluster creates a KinD cluster given the path to use to create a kubeconfig file.
+// It returns an error if something went wrong.
+func createKindCluster(kubeconfigFile string) error {
```

Comment:
Little confused, I thought that Kind was now handled outside of tests as the docs stated above?

Reply:
If you use `TESTDISTRO=kind` a kind cluster will be created

Reply:
copy must have misunderstood that part from the docs, makes sense though.